src/sagemaker/clarify.py

```diff
@@ -2092,12 +2312,26 @@ def explainability(
         explainability_config: Union[ExplainabilityConfig, List[ExplainabilityConfig]],
     ):
         """Generates a config for Explainability"""
+        # determine if this is a timeseries explainability case by checking
+        # if *both* TimeSeriesDataConfig and TimeSeriesModelConfig were given
+        ts_data_config_present = "time_series_data_config" in data_config.analysis_config
+        ts_model_config_present = "time_series_predictor_config" in model_config.predictor_config
+
+        if ts_data_config_present and ts_model_config_present:
```

Comment:
```
if asymmetric_shap:
    assert has_ts_data_config, "error mesage"
    asser has_model_config, "error message"
```

Reply:
Should we collect all validation logic at the start of the method and not spread it across all methods?

Reply:
Will address today.

Reply:
> ```
> if asymmetric_shap:
>     assert has_ts_data_config, "error mesage"
>     asser has_model_config, "error message"
> ```
Thought:

I think we should future proof and actually leave it as it is. If we ever decide that we want to accept more `ExplainabilityConfig` types than just `AsymmetricSHAPConfig` or multiple types at once like we do for tabular use case then this approach will not work, whereas the current one likely will.

Reply:
> If we ever decide that we want to

We can decide that when this time comes. For instance, it can be `if asymmetric_shap or other_ts_explanation`.

Let's focus on the current problems